src/router_core/router_core_private.h

```diff
@@ -439,10 +439,28 @@ typedef struct qdr_connection_work_t {
 ALLOC_DECLARE(qdr_connection_work_t);
 DEQ_DECLARE(qdr_connection_work_t, qdr_connection_work_list_t);
 
+struct qdr_connection_info_t {
+    const  char                *container;
+    char                       *sasl_mechanisms;
+    char                       *host;
+    bool                        is_encrypted;
+    char                       *ssl_proto;
+    char                       *ssl_cipher;
+    char                       *user;
+    bool                        is_authenticated;
+    bool                        opened;
+    const char                 *dir;
```

Comment:
I think dir and role should use the enumerated types rather than arbitrary strings.